datasource/src/main/scala/quasar/blobstore/azure/rx.scala

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2014–2018 SlamData Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package quasar.blobstore.azure
+
+import slamdata.Predef._
+
+import cats.effect._
+import cats.implicits._
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.numeric.Positive
+import fs2.{RaiseThrowable, Stream}
+import fs2.concurrent.Queue
+import io.reactivex.{Flowable, Single, SingleObserver}
+import io.reactivex.observers.DisposableSingleObserver
+import io.reactivex.subscribers.ResourceSubscriber
+
+object rx {
+  final class AsyncSubscriber[A](cb: Either[Throwable, Option[A]] => Unit) extends ResourceSubscriber[A] {
```

Comment:
I think the `DefaultSubscriber` would be preferable here, as we don't have any RX resources to add to the lifecycle of the subscription.

Additionally, for this to work with backpressure, we should only request one event at a time to allow the consumer to control demand.

As-is, this requests an unbounded number of events from the producer.

Reply:
yeah, I don't think there's any substantial difference between current impl and previous one (because the emitter isn't backpressure aware).

I'll look into supporting backpressure but like to do that as a separate issue (opened ch3139 for that) if that's ok with you.

Reply:
I think we need to change something now as with the bounded queue, the producer can produce more events than can be handled, so I'm not sure what will happen in that case. We may just end up with a bunch of suspended continuations hanging around.

I think a basic implementation would be fairly straightforward, something like
```scala
final class AsyncSubscriber[A](cb: Either[Throwable, Option[A]] => Unit) extends DefaultSubscriber[A] {
  override def onStart(): Unit =
    request(1)

  def onNext(a: A): Unit = {
    cb(Right(Some(a)))
    request(1)
  }

  def onError(t: Throwable): Unit =
    cb(Left(t))

  def onComplete: Unit =
    cb(Right(None))
}
```
in the future, we could add support for subscription cancellation when the stream halts. For now, I think this will at least keep everything well-behaved.